torchvision/prototype/transforms/kernels/_geometry.py

```diff
@@ -16,11 +16,17 @@
 def horizontal_flip_bounding_box(
     bounding_box: torch.Tensor, *, format: features.BoundingBoxFormat, image_size: Tuple[int, int]
 ) -> torch.Tensor:
+    shape = bounding_box.shape
+
     bounding_box = convert_bounding_box_format(
-        bounding_box, old_format=format, new_format=features.BoundingBoxFormat.XYXY, copy=True
-    )
-    bounding_box[..., (0, 2)] = image_size[1] - bounding_box[..., (2, 0)]
-    return convert_bounding_box_format(bounding_box, old_format=features.BoundingBoxFormat.XYXY, new_format=format)
+        bounding_box, old_format=format, new_format=features.BoundingBoxFormat.XYXY
+    ).view(-1, 4)
+
+    bounding_box[:, [0, 2]] = image_size[1] - bounding_box[:, [0, 2]]
```

Comment:
Whenever you do in-place modifications like that, make sure you have cloned the tensor prior. The above method seems not to clone in specific conditions. Check the code-base to confirm we are good for the rest.

We will also later on require a test to check explicitly for inplace modifications.